test/waittask.spec.ts

```diff
@@ -28,6 +29,12 @@ describe('waittask specs', function () {
   setupTestPageAndContextHooks();
 
   describe('Page.waitFor', function () {
+    /* This method is deprecated but we don't want the warnings showing up in
+     * tests. Until we remove this method we still want to ensure we don't break
+     * it.
+     */
+    beforeEach(() => sinon.stub(console, 'warn').callsFake(() => {}));
```

Comment:
when is the stub restored?

Reply:
Automatically after each test: https://github.com/puppeteer/puppeteer/blob/main/test/mocha-utils.ts#L273